pubsub/gcloud/aio/pubsub/subscriber.py

```diff
@@ -99,6 +100,20 @@ async def acker(subscription: str,
                 log.warning(
                     'Ack request failed, better luck next batch', exc_info=e)
                 metrics_client.increment('pubsub.acker.batch.failed')
+
+                is_response_error = isinstance(
+                    e,
+                    aiohttp.client_exceptions.ClientResponseError)
+                if is_response_error and e.status == 400:  # type: ignore # pylint: disable=no-member
+                    log.warning(
+                        'Ack error is unrecoverable, dropping the whole batch')
+                    for i in ack_ids:
+                        asyncio.ensure_future(
+                            subscriber_client.acknowledge(subscription,
+                                                          ack_ids=[i])
```

Comment:
I think this code makes sense and is a good tradeoff between batching for throughput and ensuring valid `ack_id`s get acked without being held back by one or more bad IDs, but does this constitute dropping the whole batch? Maybe I'm just nitpicking on wording but some IDs may be successfully acknowledged and others dropped, making it a "partial drop". I would personally be a bit confused by the warning message.

Again, functionally it makes sense but could the warning align with the behaviour? Something like `Ack error is unrecoverable, one or more messages may be dropped`? (and likewise for the nacker)

Reply:
I agree with the wording. We're not really dropping the batch, right? We're giving each ack id another chance (if I'm reading this correctly)

Reply:
My reasoning was that from `Acker` point of view it bears no responsibility for this batch from now on. Sure we'll make a final effort to maybe ack what can be acked, but we won't monitor the result, not will we catch or log any exceptions.

But suggested wording is more precise anyway, so sure why not